doc/design/audit-trail-mvp.md

```diff
@@ -0,0 +1,156 @@
+# Audit Trail MVP
+
+NIH has required that DANDI Archive generate audit logs while leaving undefined
+what that means. We are choosing to fulfill the requirement with a feature that
+indicates to Dandiset owners and admins the sequence of significant actions
+taken to modify Dandisets, along with identification of who performed each
+action.
+
+The major motivation is to inform owners about how their Dandisets have changed
+when a given owner did not make the change and wishes to know who to speak to
+about those changes. In essence, it is a tool to support effective
+collaboration.
+
+## Requirements
+
+1. **Persistent tracking of significant actions.** Whenever a *significant
+   action* (see list [below](#significant-actions)) occurs successfully, the archive will generate a
+   persistent record of that action, including the following data:
+    - The date and time
+    - The authorized user who triggered the action (note that this may be an actual user or the system itself for internal automated actions)
+    - The ID of the Dandiset on which the action occurs
+    - Event-specific data about the action itself
+
+2. **Display of audit log (to owners and admins only).** The Dandiset landing
```

Comment:
Don't we want to gain API (hence a separate item before this one) to query the audit log and then "display" would be based on that?

Reply:
That's not an absolute requirement. The API to deliver the display results could be private, for example. And admin users will, for instance, be able to see audit records using the admin console.

If we actually want *public* API to be a part of this, we should discuss why we want that, and then we'll need to add it to the requirements. But my understanding is that the admins want to be able to see the audit events for debugging etc., and the display could be offered to Dandiset owners as a feature.

To avoid extending this discussion (and the implementation phase) further, I would argue for adding a public API later if needed for some reason.

Reply:
given the second sentence above in the opening paragraph, wouldn't an implementation detail of this be an API end point? since owners only interact via the api.

Reply:
Yes, I think that's what I was trying to get at: implementation details are not requirements; if we want to have a public API for audit data, that would be a requirement, and it would mean we need to do more design to get it right. I'm arguing for allowing whatever API is necessary for this MVP to remain private, and therefore not trigger any extra design before we can get to implementation.

If, in the future, we decide it needs to be a public API, we can do the work at that point to make it public. In the meantime, owners and admins will be able to browser the audit logs in-app, and admins will additionally be able to search/view/etc. from the admin console.

Reply:
While the question of whether the display of the audit log in the web interface should be based on some API is an implementation detail that has no place in a requirements doc, I think there should still be some API endpoint(s) for accessing ths logs (regardless of how that ends up relating to the web interface), as otherwise `backups2datalad` won't be able to benefit from this.

Reply:
I could argue that "Display of ... on DLP" is an implementation detail of a "Provide access to ".
I think that overall hiding the need for designing API access could foster implementing some solution which would workaround that need and thus becoming hard to change later on.  If the concern is not making API private, could be stated explicitly  that API in this MVP would remain "admin-only" but the point should be that there would be an API end-point and that "Display on DLP" should use that API as a requirement to avoid possible ad-hoc DLP-only solution.